server/backends/migration_cache/migration_cache.go

```diff
@@ -196,6 +207,13 @@ func (mc *MigrationCache) Get(ctx context.Context, d *repb.Digest) ([]byte, erro
 		}
 	}
 
+	// Enqueue non-blocking copying
+	select {
+	case mc.copyChan <- d:
```

Comment:
We should also capture the context passed to `Get` for the background copy (ExtendContextForFinalization might be useful here). The context will contain auth information which is necessary for the cache code to work correctly.

Reply:
Ah good point - fixed